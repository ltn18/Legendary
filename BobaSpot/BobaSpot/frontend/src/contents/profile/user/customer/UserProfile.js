import React, { useState } from 'react';

import UserInfo from './UserInfo';
import UserComments from './UserComments';

const UserProfile = () => {
    return (
        <div
            style={{
                height: '100vh',
                display: 'flex',
            }}
        >
            <div
                style={{ width: '30%' }}
                data-testid="user-info"
            >
                <UserInfo />
            </div>
            <div
                style={{
                    width: '70%',
                }}
                data-testid="user-comments"
            >
                <UserComments />
            </div>
        </div>
    )
}

export default UserProfile;