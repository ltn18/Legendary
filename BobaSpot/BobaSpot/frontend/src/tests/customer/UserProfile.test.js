import React from 'react';
import { render, screen } from '@testing-library/react';
import UserProfile from "../../contents/profile/user/customer/UserProfile";

describe('UserProfile', () => {
    it('renders UserInfo and UserComments components', () => {
      const { getByTestId } = render(<UserProfile />);
      const userInfoElement = getByTestId('user-info');
      const userCommentsElement = getByTestId('user-comments');
      expect(userInfoElement).toBeInTheDocument();
      expect(userCommentsElement).toBeInTheDocument();
    });
  });