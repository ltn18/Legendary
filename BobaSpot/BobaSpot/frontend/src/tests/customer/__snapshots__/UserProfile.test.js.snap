// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows a user info panel and a user comments panel 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "height": "100vh",
    }
  }
>
  <div
    style={
      Object {
        "width": "25%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "background": "rgba(255, 255, 255, 0.2)",
            "display": "flex",
            "height": "25%",
            "justifyContent": "space-between",
            "padding": 10,
          }
        }
      >
        <div
          style={
            Object {
              "color": "white",
              "width": "50%",
            }
          }
        >
          <div
            className="ant-image css-dev-only-do-not-override-1e3x2xa"
            onClick={[Function]}
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <img
              className="ant-image-img"
              onLoad={[Function]}
              src="https://i.ytimg.com/an/zlwQERpksnw/14720571135996419329_mq.jpg?v=6286689c"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
            <div
              className="ant-image-mask"
              style={
                Object {
                  "display": undefined,
                }
              }
            >
              <div
                className="ant-image-mask-info"
              >
                <span
                  aria-label="eye"
                  className="anticon anticon-eye"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                    />
                  </svg>
                </span>
                Preview
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "color": "white",
              "paddingLeft": 30,
              "width": "50%",
            }
          }
        >
          <h1
            style={
              Object {
                "color": "black",
                "fontSize": 30,
              }
            }
          >
             
          </h1>
          <div
            style={
              Object {
                "color": "black",
                "fontSize": 20,
              }
            }
          >
            <i
              style={Object {}}
            >
              username:
            </i>
             
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "background": "white",
            "height": "75%",
            "padding": 10,
          }
        }
      >
        <div
          className="ant-space css-dev-only-do-not-override-1e3x2xa ant-space-vertical"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="ant-space-item"
            style={
              Object {
                "marginBottom": 8,
              }
            }
          >
            <button
              className="ant-btn css-dev-only-do-not-override-1e3x2xa ant-btn-default ant-btn-lg ant-btn-block"
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d7d7d7",
                  "borderRadius": 0,
                  "boxShadow": "0px 4px 4px rgba(0, 0, 0, 0.25)",
                  "color": "black",
                }
              }
              type="button"
            >
              <span>
                Change user's info
              </span>
            </button>
          </div>
          <div
            className="ant-space-item"
            style={Object {}}
          >
            <button
              className="ant-btn css-dev-only-do-not-override-1e3x2xa ant-btn-default ant-btn-lg ant-btn-block ant-btn-dangerous"
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 0,
                  "boxShadow": "0px 4px 4px rgba(0, 0, 0, 0.25)",
                  "color": "#ff0e0e",
                }
              }
              type="button"
            >
              <span>
                Logout
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "width": "75%",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#FDD0CF",
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "padding": 20,
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
